src/web/components/StickyBottomBanner/StickyBottomBanner.tsx

```diff
@@ -18,63 +22,96 @@ const getBannerLastClosedAt = (key: string): string | undefined => {
     return (item && JSON.parse(item).value) || undefined;
 };
 
+const buildNewCmpBannerConfig = () => {
+    return {
+        id: 'cmpUi',
+        canShow: () => willShowCMP().then((result) => ({ result: !!result })),
+        // New CMP is not a react component and is shown outside of react's world
+        // so render nothing if it will show
+        show: () => null,
+        timeoutMillis: null,
+    };
+};
+
+const buildOldCmpBannerConfig = (CAPI: CAPIBrowserType) => {
+    return {
+        id: 'cmpUi',
+        canShow: () =>
+            shouldShowOldCMP().then((shouldShowOld) => ({
+                result: shouldShowOld && CAPI.config.cmpUi,
+            })),
+        show: () => CMP,
+        timeoutMillis: null,
+    };
+};
+
+const buildReaderRevenueBannerConfig = (
+    CAPI: CAPIBrowserType,
+    isSignedIn: boolean,
+    asyncCountryCode: Promise<string>,
+) => {
+    return {
+        id: 'reader-revenue-banner',
+        canShow: () =>
+            canShowRRBanner({
+                remoteBannerConfig: CAPI.config.remoteBanner,
+                isSignedIn,
+                asyncCountryCode,
+                contentType: CAPI.contentType,
+                sectionName: CAPI.sectionName,
+                shouldHideReaderRevenue: CAPI.shouldHideReaderRevenue,
+                isMinuteArticle: CAPI.pageType.isMinuteArticle,
+                isPaidContent: CAPI.pageType.isPaidContent,
+                isSensitive: CAPI.config.isSensitive,
+                tags: CAPI.tags,
+                contributionsServiceUrl: CAPI.contributionsServiceUrl,
+                alreadyVisitedCount: getAlreadyVisitedCount(),
+                engagementBannerLastClosedAt: getBannerLastClosedAt(
+                    'engagementBannerLastClosedAt',
+                ),
+                subscriptionBannerLastClosedAt: getBannerLastClosedAt(
+                    'subscriptionBannerLastClosedAt',
+                ),
+                switches: {
+                    remoteSubscriptionsBanner: !!CAPI.config
+                        .remoteSubscriptionsBanner,
+                },
+            }),
+        /* eslint-disable-next-line react/jsx-props-no-spreading */
+        show: (meta: any) => () => <ReaderRevenueBanner {...meta} />,
```

Comment:
Note that by omitting the optional `timeoutMillis` parameter here we're automatically opting in the reader revenue banner to the 2 second timeout rule, which may not be what we want. We can opt out of this by specifying `timeoutMillis: null`.

Reply:
You're sending timeout events to ophan so I'm ok with this (it behaves the same as in frontend). We can monitor timeouts and reassess later